be2-scala/src/main/scala/ch/epfl/pop/storage/DiskStorage.scala

```diff
@@ -29,24 +29,40 @@ class DiskStorage(val databaseFolder: String = DiskStorage.DATABASE_FOLDER) {
 
 
   @throws [org.iq80.leveldb.DBException]
-  def read(key: String): String = {
-    new String(db.get(key.getBytes(StandardCharsets.UTF_8)), StandardCharsets.UTF_8)
+  def read(key: String): Option[String] = {
+    db.get(key.getBytes(StandardCharsets.UTF_8)) match {
+      case null => None
+      case bytes => Some(new String(bytes, StandardCharsets.UTF_8))
+    }
   }
 
   @throws [org.iq80.leveldb.DBException]
-  def write(keyValues: (String, String)*): Unit = {
-    val batch: WriteBatch = db.createWriteBatch()
+  def write(key: String, value: String): Unit = {
+    db.put(key.getBytes(StandardCharsets.UTF_8), value.getBytes(StandardCharsets.UTF_8))
+  }
 
-    try {
-      for (kv <- keyValues) {
-        batch.put(kv._1.getBytes(StandardCharsets.UTF_8), kv._2.getBytes(StandardCharsets.UTF_8))
-      }
+  @throws [org.iq80.leveldb.DBException]
+  def write(keyValues: (String, String)*): Unit = keyValues.size match {
```

Comment:
do we really need the two implementations + matching ? can't we have a very simple generalized implementation ? 😄